lib/THC/THCTensor.c

```diff
@@ -599,7 +599,7 @@ void THCudaTensor_freeCopyTo(THCState *state, THCudaTensor *self, THCudaTensor *
 static void THCudaTensor_rawInit(THCState *state, THCudaTensor *self)
 {
   self->refcount = 1;
-  self->storage = NULL;
+  self->storage = THCudaStorage_new(state);
```

Comment:
Sorry, you lost me here. Reading the code, it looks like it will return `0` (aka `NULL`) in both cases. I'm not sure what it means for a C function to return `nil`.

    float *THCudaTensor_data(THCState *state, const THCudaTensor *self)
    {
      if(self->storage)
        return (self->storage->data+self->storageOffset);
      else
        return NULL;
    }